org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/hotspots/SecurityHotspotsHandlerServer.java

```diff
@@ -150,14 +152,19 @@ public void shutdown() {
 
     @Override
     public void handle(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context) throws HttpException, IOException {
+      boolean trustedServer = Optional.ofNullable(request.getHeader("Origin"))
```

Comment:
FYI @damien-urruty-sonarsource I tried to write an IT for this, but `java.net.HttpURLConnection` would not let me change the `Origin` header. We would need to either set a (`sun.*`) system property to allow it, or migrate the tests to e.g Commons HTTP client. Not completely sure it's worth the investment.